docs/development/local_setup.md

```diff
@@ -0,0 +1,85 @@
+```
+
+Make sure to set your `$GOPATH` environment variable properly (conventionally, it points to `$HOME/go`).
+
+For your convenience, you can add the `bin` directory of the `$GOPATH` to your `$PATH`: `PATH=$PATH:$GOPATH/bin`, but it is not necessarily required.
+
+In order to perform linting on the Go source code, install [Golint](https://github.com/golang/lint):
+
+```bash
+$ go get -u github.com/golang/lint/golint
+```
+
+[Dep](https://github.com/golang/dep) is used for managing Golang package dependencies. Install it:
+```bash
+$ brew install dep
+```
+
+### Installing `git`
+We use `git` as VCS which you need to install:
+
+```bash
+$ brew install git
+```
+
+### Installing `Docker` (Optional)
+In case you want to build Docker images for the Node Controller Manager you have to install Docker itself. We recommend using [Docker for Mac OS X](https://docs.docker.com/docker-for-mac/) which can be downloaded from [here](https://download.docker.com/mac/stable/Docker.dmg).
+
+### Setup `Docker Hub` account (Optional)
+Create a Docker hub account at [Docker Hub](https://hub.docker.com/) if you don't already have one.
+
+## Local development
+
+:warning: Before you start developing, please ensure to comply with the following requirements:
+
+1. You have understood the [principles of Kubernetes](https://kubernetes.io/docs/concepts/), and its [components](https://kubernetes.io/docs/concepts/overview/components/), what their purpose is and how they interact with each other.
+1. You have understood the [architecture of the Node Controller Manager](../design/node_controller_manager.md)
+
+The development of the Node Controller Manager could happen by targetting any cluster. You basically need a Kubernetes cluster running over a set of machines. This cluster must be remotely connectable. You just need the [Kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/authenticate-across-clusters-kubeconfig/) file with the required access permissions attached to it.
+
+### Installing the Node Controller Manager locally
+Clone the repository from GitHub into your `$GOPATH`.
+
+```bash
+$ mkdir -p $GOPATH/src/github.com/gardener
+$ cd $GOPATH/src/github.com/gardener
+$ git clone git@github.com:gardener/node-controller-manager.git
+$ cd node-controller-manager
+```
+
+### Prepare the cluster
+
+- Connect to the remote kubernetes cluster where you plan to deploy the Node Controller Manager using the kubectl. Make the KUBECONFIG environment variable point to the yaml file containing your cluster info.
+- Now, create the required CRDs on the remote cluster using the following command,
+```bash
+$ kubectl apply -f kubernetes/crds.yaml
+```
+
+### Get started
+
+Please create a `dev` directory and copy the Kubeconfig of the kubernetes cluster used for development purposes to `dev/kubeconfig.yaml`.
+
+- For your convenience, we have created a rule in the `Makefile` which will automatically start the Node Controller Manager with development settings:
```

Comment:
s/For your convenience, we have created a rule in the/There is a rule `dev` in the/